### PR TITLE
Close SQL statements and results after being used.

### PR DESCRIPTION
**Pull request by joe32344**
Close SQL statements and results sets once they have been used, this is a small change, but over time they can build up resulting in reduced performance.

### CODE REVIEW

File: src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java [hunk 1]
```diff
@@ -550,6 +554,7 @@ private void savePlayerData(String playerID, PlayerData playerData)
             insertStmnt.setInt(3, playerData.getAccruedClaimBlocks());
             insertStmnt.setInt(4, playerData.getBonusClaimBlocks());
             insertStmnt.executeUpdate();
+            insertStmnt.close();
```
Comment (Jikoo) [inline src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java:557]: Unnecessary, try-with-resources closes this.

File: src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java [hunk 2]
```diff
@@ -512,6 +515,7 @@ PlayerData getPlayerDataFromStorage(UUID playerID)
                 playerData.setAccruedClaimBlocks(results.getInt("accruedblocks"));
                 playerData.setBonusClaimBlocks(results.getInt("bonusblocks"));
             }
+            results.close();
```
Comment (Jikoo) [inline src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java:518]: Unnecessary, try-with-resources closes statement, which closes this.

File: src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java [hunk 3]
```diff
@@ -661,12 +666,15 @@ protected int getSchemaVersionFromStorage()
...
+                results.close();
                 return 0;
             }
             //otherwise return the value that's in the table
             else
             {
-                return results.getInt("version");
+                int ver = results.getInt("version");
+                results.close();
+                return ver;
```
Comment (Jikoo) [inline src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java:677]: Same with all of these, try-with-resources closes statement, which closes this.

File: src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java [hunk 4]
```diff
@@ -411,6 +411,9 @@ else if (parentId == -1)
             statement.execute("DELETE FROM griefprevention_claimdata WHERE id='-1';");
         }
 
+        results.close();
+        statement.close();
```
Comment (Jikoo) [inline src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java:415]: Good, but there are a couple missed cases earlier in the method where reassignment can leak unclosed resources.

**Review by Jikoo — CHANGES_REQUESTED** [review]
Does fix some of initialization's leaks, but overall mostly unnecessary due to how a [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) works.

File: src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java (hunk 4 above)
Comment (RoboMWM) [reply src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java:415]: did he just mark these all as resolved without doing anything

**Review by RoboMWM — CHANGES_REQUESTED** [review]

File: src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java (hunk 4 above)
Comment (Jikoo) [reply src/main/java/me/ryanhamshire/GriefPrevention/DatabaseDataStore.java:415]: Yes

**Pull request closed without merging**